ClientContext.cs

```diff
@@ -60,49 +58,29 @@ public static ClientContext Build(string tenantId, string objectId, string appId
         /// </summary>
         /// <param name="certificateThumbprint"></param>
         /// <returns></returns>
-        public static Task<ServiceClientCredentials> GetServiceCredentialsAsync( string tenantId, string applicationId, string appSecret )
+        public AzureCredentials GetAzureCredentialsAsync(string clientId, string clientSecret, string tenantId, AzureEnvironment environment)
         {
-            if (_servicePrincipalCredential == null)
-            {
-                _servicePrincipalCredential = new ClientCredential(applicationId, appSecret);
-            }
-
-            return ApplicationTokenProvider.LoginSilentAsync(
-                tenantId,
-                _servicePrincipalCredential, 
-                ActiveDirectoryServiceSettings.Azure,
-                TokenCache.DefaultShared);
+            return SdkContext.AzureCredentialsFactory.FromServicePrincipal(clientId, clientSecret, tenantId, environment); 
         }
 
-        /// <summary>
-        /// Generic ADAL Authentication callback
-        /// </summary>
-        public static async Task<string> AcquireTokenAsync(string authority, string resource, string scope)
```

Comment:
To track 2, there is new SecretClient instead of KeyVaultClient to use. this AcquireTokenAsync() is used to create KeyVaultClient in KeyvaultSampleBase delegate. So this is decided to remove